### PR TITLE
Build for Linux on GitHub Actions

### PR DESCRIPTION
**Pull request by Dahlgren**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,6 +3,33 @@ name: CI
...
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Install dependencies
+        run: |
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+          sudo apt-get update
+          sudo apt-get install -y gcc-11 g++-11
+          sudo apt-get remove -y libtbb-dev
```
Comment (Dahlgren) [inline .github/workflows/ci.yml:19]: Build fails if libtbb is installed. Should probably be fixed in cmake...

File: src/optimizer/optimizerModuleBase.cpp [hunk 2]
```diff
@@ -272,10 +272,10 @@ OptimizerModuleBase::Node OptimizerModuleBase::nodeFromAST(const astnode& input)
             std::vector<std::vector<astnode>::const_iterator> lastChildren;
 
             while (!statements->children.empty()) {
-                auto& lastChild = (statements->children.end() - 1);
+                auto lastChild = (statements->children.end() - 1);
 
                 lastToken = lastChild->token.offset + lastChild->token.contents.size();
-                statements = lastChild._Ptr;
+                statements = &lastChild->children[0];
```
Comment (dedmen) [inline src/optimizer/optimizerModuleBase.cpp:278]: I'm pretty sure these are not the same thing

Comment (Dahlgren) [reply src/optimizer/optimizerModuleBase.cpp:278]: It breaks Windows at runtime so nope 😸

**Comment by jonpas** [comment]
This is ready to merge.

**Comment by Tuupertunut** [comment]
On Windows ArmaScriptCompiler expects that the project is built in the root of a drive, such as the P-drive. How does this work on Linux? Does the project need to be in the filesystem root directory?

**Comment by jonpas** [comment]
@dedmen merge so we can update in HEMTT?

**Pull request merged**